### PR TITLE
Allow navigation instead of popup in subrecord lists

### PR DESCRIPTION
**Pull request by sleidig**
see commits: 
- refactored to generalize similar components
- added option to configure subrecord table to navigate instead of showing popup (suggestions for a name for this flag welcome! `clickMode` doesn't seem very expressive ...)

### CODE REVIEW

File: src/app/core/entity-components/entity-subrecord/entity-subrecord/entity-subrecord.component.ts [hunk 1]
```diff
@@ -379,16 +384,35 @@ export class EntitySubrecordComponent<T extends Entity>
    * Show one record's details in a modal dialog (if configured).
    * @param row The entity whose details should be displayed.
    */
-  rowClick(row: TableRow<T>) {
+  onRowClick(row: TableRow<T>) {
+    //if (row.formGroup && !row.formGroup.disabled) {
+
```
Comment (TheSlimvReal) [inline src/app/core/entity-components/entity-subrecord/entity-subrecord/entity-subrecord.component.ts:389]:
```suggestion
```

File: src/app/core/entity/entity-config.service.ts [hunk 2]
```diff
@@ -123,4 +124,9 @@ export interface EntityConfig {
    * icon used to visualize the entity type
    */
   icon?: string;
+
+  /**
+   * base route of views for this entity type
+   */
+  route?: string;
```
Comment (TheSlimvReal) [inline src/app/core/entity/entity-config.service.ts:131]: Great idea. Could we use the chance to also use this for all entities? At the moment we have some places where we still create the route through `ENTITY_TYPE + ':' + id` (e.g. in the `EntityBlockComponent`) which is rather dangerous as it requires the route to be equal to the entity type.
With this approach we have much more flexibility and can make the routes however we want.

File: src/app/core/config/config-fix.ts [hunk 3]
```diff
@@ -644,9 +644,10 @@ export const defaultJsonConfig = {
           "components": [
             {
               "title": $localize`:Title inside a panel:School History`,
-              "component": "PreviousSchools",
+              "component": "ChildSchoolOverview",
               "config": {
                 "single": true,
+                "clickMode": "navigate",
```
Comment (TheSlimvReal) [inline src/app/core/config/config-fix.ts:650]:
```suggestion
```
This is not working in the demo.

**Review by TheSlimvReal — CHANGES_REQUESTED** [review]
Seems to be working well. Thanks for taking this up. 

Playing with the app I discovered the following styling issue:
- [ ] Somehow the tables in the details views are now missing some top-margin (as compared to the currently deployed demo) (might be an individual change though)

File: src/app/child-dev-project/schools/child-school-overview/child-school-overview.component.ts [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    switch (entity?.getConstructor()?.ENTITY_TYPE) {
+      case Child.ENTITY_TYPE:
+        return "child";
+      case School.ENTITY_TYPE:
+        return "school";
+    }
+  }
+
+  async loadData() {
+    if (!this.entity.getId() || this.entity.getId() === "" || !this.mode) {
```
[the remark this answers is not in the document]
Comment (sleidig) [reply src/app/child-dev-project/schools/child-school-overview/child-school-overview.component.ts:80]:
```suggestion
    if (!this.mode) {
```
to be honest - I have no idea. These have been in one of the previous component implementations but I think we can assume a valid entity is set here

File: src/app/core/config/config-fix.ts (hunk 3 above)
Comment (sleidig) [reply src/app/core/config/config-fix.ts:650]: :+1: thanks. forgot this here from testing, apparently

File: src/app/core/entity/entity-config.service.ts (hunk 2 above)
Comment (sleidig) [reply src/app/core/entity/entity-config.service.ts:131]: good point actually. Done.

**Review by TheSlimvReal — APPROVED** [review]
Great, works fine now

**Pull request merged**